### DISCUSSION TITLE
Deploying BERTopic on AWS

### DISCUSSION THREAD

**Discussion by trivikram18** [MaartenGr/BERTopic#1370] [Q&A] [answered]
Hello All,

I am deploying BERTopic on AWS via Docker and AWS ECR and calling it from a lambda function.
I am running into an issue as follows ...

[Errno 30] Read-only file system: '/home/sbx_user1051'

The code runs fine locally (windows 10)
Did anyone run into this issue? Any help would be great.

Thanks
Trivikram

**Comment by MaartenGr** [comment]
Could you give the full error log as well as the training code? From the message it might have something to do with saving the model but it is difficult to say without seeing the entire code and log.

**Comment by trivikram18** [comment]
Hi Maarten, 

Thanks for the revert. Please see the following screenshot for the full error log.

![249685360-2aa7360b-a059-46d8-a6eb-df740ccd4cf7](https://github.com/MaartenGr/BERTopic/assets/10544722/d85cf69f-6c16-456a-af0a-5e657c7b5273)

As per the documentation from AWS, we can only save in /tmp/ folder and the rest of the installation folders, etc are all read-only. 
So if one of the sub-models is getting saved then that might be the area where the error might be occurring. 
Where does the code save the sub-model? Maybe I can change that path to /tmp/ and use it. 

Thanks
Trivikram

**Reply by MaartenGr** [reply]
With sentence-transformers, it is also possible to locally save the model and then load it from the local environment. This might prevent the issue that you are having. For example, you could download [all-MiniLM-L6-v2](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2) locally, upload it to AWS and then load it from there.

**Comment by trivikram18** [comment]
I am already saving the model files for all-MiniLM-L6-v2 locally and using the saved model, so I believe it's not to do with this. 
Does the code try to save the UMAP and HDBSCAN models internally and use them?

**Reply by MaartenGr** [reply]
If I am not mistaken, only the embedding model is downloaded and saved if you load one in from the web. Other than that, there should be no saving of models during training. Could you share your full code? That will make it a bit easier to debug. Also, it does seem like the model is being downloaded and saved so I am not entirely sure whether you loaded in your local model correctly.

**Comment by trivikram18** [comment]
Thanks, Maarten. Do you think we could take this offline and post the solution here?

**Reply by MaartenGr** [reply]
Sure, feel free to send a mail if you need help. Although I generally am quicker to respond here.

**Comment by trivikram18** [comment, marked as answer]
Here is the solution to the issue mentioned:

I was saving the model from sentence_transformers wrong when saving to the local drive. Here is the correct way ... 

```python
from sentence_transformers import SentenceTransformer
sentence_model = SentenceTransformer("all-MiniLM-L6-v2")
```

Then, save that model locally to a folder of your choice:

```python
sentence_model.save("my_model")
```

You can then upload that folder to your AWS instance and load it back as follows:

```python
from sentence_transformers import SentenceTransformer
sentence_model = SentenceTransformer("my_model")
```

Thanks a lot, Maarten for your guidance and help on this.

**Reply by MaartenGr** [reply]
No problem! Glad I could be of help 😄
